### PR TITLE
Fix warnings, clean namings and overall improvement

### LINKED ISSUE #312: Naming convention for mutated reads rate
Decide if we want to update the term for mutated reads rate, and call it mutation burden or whatever we prefer?

### LINKED ISSUE #332: Fix filenames of site comparison results
Right now these are all the files that are outputted from the site comparison analysis, it would be great if we could make sure that the filenames are all different and as self-explanatory as possible.

this is when not activating the omega_multi option
```
sitecomparison/all_samples.aminoacid_change.comparison.tsv.gz
sitecomparisongloballoc/all_samples.global_loc.aminoacid_change.comparison.tsv.gz
sitecomparisongloballocmulti/all_samples.aminoacid_change.comparison.tsv.gz
sitecomparisonmulti/all_samples.aminoacid_change.comparison.tsv.gz
```

same for aminoacid and site level values

### LINKED ISSUE #334: Relative mutability outputs misnamed
Relative mutability outputs should be named ending with `.tsv.gz`, rather than `.gz`.  For example, instead of outputs `relativemutability/NBB_0297_FIU_01_SONzL_EL20220528_6zS20220912zp20250711zdeepUMI.all.adjusted.gz` and `relativemutability/NBB_0297_FIU_01_SONzL_EL20220528_6zS20220912zp20250711zdeepUMI.all.adjusted.gz.tbi`, the outputs should be named `relativemutability/NBB_0297_FIU_01_SONzL_EL20220528_6zS20220912zp20250711zdeepUMI.all.adjusted.tsv.gz` and `relativemutability/NBB_0297_FIU_01_SONzL_EL20220528_6zS20220912zp20250711zdeepUMI.all.adjusted.tsv.gz.tbi`.

### LINKED ISSUE #344: revise contamination outputs
csv files for contamination outputs includes shared germline mutations. Is it what we want? Probably it would be more informative to output only non-shared germline mutations present in other sample as somatic.
Idea for the improvement: in the file with contamination also output numbers of mutations leading to the obtained proportion (could be useful for some cases where really high proportion comes from very low numbers) and the source of the contamination.

### LINKED ISSUE #402: fix nextflow warnings
- [x] Channel to channel
- [x] it without prior definitions

### LINKED ISSUE #415: Add computation of relative coverage of the gene
Add a step that takes the information from the consensus panel definition and the information coming from the APIs on the total number of positions of the gene and reports how much of the total length of the gene is included in the consensus panel.

This will be informative to decide if one can trust the information coming from dNdScv since there we are assuming that the full gene is covered (aka mutations are allowed in the entire gene) and in case it is not, we would be underestimating the dN/dS values obtained with these method.

### LINKED ISSUE #416: All mutations being filtered from project
Hi all (@FerriolCalvet , especially)

I'm trying to run about 100 samples through deepCSA using a method that has worked before.  I'm using `-r v1.0.1` (commit hash e7ace4480fcf095d99972016aa94ac90a5d1a4ed), and my run command is 
```bash
#!/bin/bash

set -Eeuo pipefail

# Nextflow Version
NXF_VER=25.04.2

# Nextflow Configuration File
NXF_CONFIG=/research/labs/kennedy/bioinf/nextflow.config

# Workflow to Run (e.g. GitHub repository)
WORKFLOW_REPO="bbglab/deepCSA"

# Workflow-Specific Options
run_config="PR024_20260128_deepCSA_config.csv"
features_params="PR024_featureParams.yaml"

# not working, add later
#    --oncodrive3d true \
#    --omega_mutabilities true \
#    --omega_globalloc true \

# Run the workflow
NXF_VER=$NXF_VER \
nextflow \
    run \
    ${WORKFLOW_REPO} \
    -with-report ${run_config}_postproc.html \
    -c ${NXF_CONFIG} \
    -params-file ${features_params} \
    --input ${run_config} \
    --outdir ${run_config}_deepCSA_resultsClonal/ \
    --fasta /mnt/labs/kennedy/bioinf/ref/hg38/hg38.fa \
    --cosmic_ref_signatures "/mnt/labs/risques/bioinf/COSMIC/COSMIC_v3.4_SBS_GRCh38.txt" \
    --wgs_trinuc_counts "/mnt/labs/risques/bioinf/deepCSA_resources/assets/trinucleotide_counts/trinuc_counts.homo_sapiens.tsv" \
    --cadd_scores "/mnt/labs/risques/bioinf/CADD/v1.7/hg38/whole_genome_SNVs.tsv.gz" \
    --cadd_scores_ind "/mnt/labs/risques/bioinf/CADD/v1.7/hg38/whole_genome_SNVs.tsv.gz.tbi" \
    --dnds_ref_transcripts "/mnt/labs/risques/bioinf/deepCSA_resources/dnds/RefCDS_human_latest_intogen.rda" \
    --dnds_covariates "/mnt/labs/risques/bioinf/deepCSA_resources/dnds/covariates_hg19_hg38_epigenome_pcawg.rda" \
    --datasets3d "/mnt/labs/risques/bioinf/oncodrive3d_resources/datasets" \
    --annotations3d "/mnt/labs/risques/bioinf/oncodrive3d_resources/annotations" \
    --domains_file "/mnt/labs/risques/bioinf/oncodrive3d_resources/o3d_pfam_parsed.tsv" \
    --vep_cache "/mnt/labs/risques/bioinf/vep_cache" \
    --vep_genome "GRCh38" \
    --vep_species "homo_sapiens" \
    --vep_cache_version 111 \
    --vep_out_format "tab" \
    --vep_params "--no_stats --cache --offline --symbol --protein --canonical --af_gnomadg --af_gnomade" \
    --vep_params_panel "--no_stats --cache --offline --symbol --protein --canonical" \
    --all_duplex_counts true \
    -profile slurm_singularity \
    --mutationrate true \
    --profileall true \
    --oncodrivefml true \
    --oncodriveclustl true \
    --omega_mutabilities true \
    --o3d_raw_vep true \
    --o3d_plot true \
    --omega  true \
    --omega_multi true \
    --omega_plot true \
    --omega_withingene true \
    --omega_autodomains true \
    --omega_autoexons true \
    --indels true \
    -process.errorStrategy ignore \
    -N kohrnb@uw.edu \
    -r v1.0.1 \
    -with-dag \
    -resume
```

What I'm getting is a bunch of errors like: 
```text
NOTE: Process `BBGTOOLS:DEEPCSA:MUT_PREPROCESSING:CLEANMUTATIONS (TBB_0422_BLO_01)` terminated with an error exit status (1) -- Error is ignored
```

which appears to originate from (looking in the working directory at `.command.log`:
```text
Subsetting MAF file...
[('FILTER', 'notcontains NM20'), ('FILTER', 'notcontains n_rich'), ('FILTER', 'notcontains cohort_n_rich_threshold'), ('FILTER', 'notcontains cohort_n_rich'), ('FILTER', 'notcontains no_pileup_support'), ('FILTER', 'notcontains low_mappability'), ('FILTER', 'notcontains not_covered'), ('FILTER', 'notcontains nanoseq_noise'), ('DEPTH', 'ge 40'), ('VAF_distorted_expanded_sq', False)]
Applying FILTER:notcontains NM20 filter implied going from 0 mutations to 0 mutations.
Traceback (most recent call last):
  File "/opt/conda/lib/python3.13/site-packages/pandas/core/indexes/base.py", line 3805, in get_loc
    return self._engine.get_loc(casted_key)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^
  File "index.pyx", line 167, in pandas._libs.index.IndexEngine.get_loc
  File "index.pyx", line 196, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 7081, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 7089, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'FILTER'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/mnt/home/kohrnb/.nextflow/assets/bbglab/deepCSA/bin/subset_maf.py", line 89, in <module>
    main()
    ~~~~^^
  File "/opt/conda/lib/python3.13/site-packages/click/core.py", line 1161, in __call__
    return self.main(*args, **kwargs)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/opt/conda/lib/python3.13/site-packages/click/core.py", line 1082, in main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.13/site-packages/click/core.py", line 1443, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/lib/python3.13/site-packages/click/core.py", line 788, in invoke
    return __callback(*args, **kwargs)
  File "/mnt/home/kohrnb/.nextflow/assets/bbglab/deepCSA/bin/subset_maf.py", line 86, in main
    subset_mutation_dataframe(sample_name, mut_file, out_maf, json_filters, req_fields, min_mutations)
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/home/kohrnb/.nextflow/assets/bbglab/deepCSA/bin/subset_maf.py", line 37, in subset_mutation_dataframe
    annotated_maf = filter_maf(raw_annotated_maf, data_tuples)
  File "/mnt/home/kohrnb/.nextflow/assets/bbglab/deepCSA/bin/utils.py", line 71, in filter_maf
    maf_df = maf_df[operators[operator](maf_df[col], value)]
                                        ~~~~~~^^^^^
  File "/opt/conda/lib/python3.13/site-packages/pandas/core/frame.py", line 4102, in __getitem__
    indexer = self.columns.get_loc(key)
  File "/opt/conda/lib/python3.13/site-packages/pandas/core/indexes/base.py", line 3812, in get_loc
    raise KeyError(key) from err
KeyError: 'FILTER'
```

All samples were run through deepUMIcaller, revision dev (commit hash 74cdb8f484f8815ba8276679a57f8eec4381e18a) before this.  

On looking at the input file there, there don't seem to be any mutations in it.  Any idea what I could change in my parameters to avoid this issue?  Or is there some other potential issue that could be causing this?

### PR DESCRIPTION
**Pull request by FerriolCalvet**
## AI summary
This pull request introduces several significant changes to the pipeline's configuration and output management, focusing on improved organization of output files, clearer parameter naming, and enhanced maintainability. The most important updates include a comprehensive reorganization of output directories, renaming and consolidation of process parameters, and the addition of a dedicated configuration file for result outputs.

**Output Directory and Process Organization:**

* Output directories for many processes have been standardized and reorganized to use clearer, more descriptive paths (e.g., moving outputs into `processing_files`, `mutations`, `plots`, `qc`, etc.) to improve discoverability and maintainability. (`conf/modules.config` [[1]](diffhunk://#diff-bf809c928cb10b54d251dec8a140a2d5505150f97175d8d7b51dda9cc57971feL16-R16) [[2]](diffhunk://#diff-bf809c928cb10b54d251dec8a140a2d5505150f97175d8d7b51dda9cc57971feL241-R234) [[3]](diffhunk://#diff-bf809c928cb10b54d251dec8a140a2d5505150f97175d8d7b51dda9cc57971feL270-R262) [[4]](diffhunk://#diff-bf809c928cb10b54d251dec8a140a2d5505150f97175d8d7b51dda9cc57971feL283-L289) [[5]](diffhunk://#diff-bf809c928cb10b54d251dec8a140a2d5505150f97175d8d7b51dda9cc57971feL428-R431) [[6]](diffhunk://#diff-bf809c928cb10b54d251dec8a140a2d5505150f97175d8d7b51dda9cc57971feL574-R577)
* A new configuration file, `results_outputs.config`, was added to centralize and fine-tune the output locations and patterns for plots, QC, and annotation steps, further enhancing output consistency. (`conf/results_outputs.config` [[1]](diffhunk://#diff-2bd5559690f8e90e649827ebb80154b2362db547a22790e8bbdb412cc09f3649R1-R284) `conf/modules.config` [[2]](diffhunk://#diff-bf809c928cb10b54d251dec8a140a2d5505150f97175d8d7b51dda9cc57971feL634-R637)

**Process and Parameter Refactoring:**

* Several process names and parameters were renamed for clarity and consistency, such as changing `SUBSET*` processes to `QUERY*`, and updating parameters like `omega_withingene` to `create_subgenic_regions`. (`conf/modules.config` [[1]](diffhunk://#diff-bf809c928cb10b54d251dec8a140a2d5505150f97175d8d7b51dda9cc57971feL69-R75) [[2]](diffhunk://#diff-bf809c928cb10b54d251dec8a140a2d5505150f97175d8d7b51dda9cc57971feL86-R86) [[3]](diffhunk://#diff-bf809c928cb10b54d251dec8a140a2d5505150f97175d8d7b51dda9cc57971feL109-R109) [[4]](diffhunk://#diff-77025add79045f941e5919f20e04599deac5aee66fc502eb5fa4e743b40d835eL33-R34) [[5]](diffhunk://#diff-44f4eaadd40b2a9bf1d7f181bd236c8c186821ecea350c7fbfd5772e48a15135L20-R22)
* Deprecated or redundant configuration files and parameters were removed, including the deletion of `conf/general_files_BBG.config`. (`conf/general_files_BBG.config` [conf/general_files_BBG.configL1-L28](diffhunk://#diff-740afa1a4b7124e06d191c4df3224d6b82bd8ed77f5b6033897b340a79e4c163L1-L28))

**Schema and Documentation Updates:**

* Improved error messages for input schema fields to be more descriptive and user-friendly, and updated required fields for sample input. (`assets/schema_input.json` [assets/schema_input.jsonL28-R36](diffhunk://#diff-dc9c490fc41c232f18648f965bb3bc2ac01dce6611082a2e11594aa6907a647eL28-R36))
* Enhanced script documentation with clearer usage examples and references to input data sources. (`assets/useful_scripts/signatures_sigprofilerextractor.py` [assets/useful_scripts/signatures_sigprofilerextractor.pyL38-R40](diffhunk://#diff-b57c91313536bbc1344c0fce6874dc17a7297a201fcc976f1ba209d9a1160a2fL38-R40))

**Other Notable Changes:**

* Adjusted publishDir settings for specific processes to disable output where appropriate or to refine file patterns and save logic. (`conf/modules.config` [[1]](diffhunk://#diff-bf809c928cb10b54d251dec8a140a2d5505150f97175d8d7b51dda9cc57971feL177-L183) [[2]](diffhunk://#diff-bf809c928cb10b54d251dec8a140a2d5505150f97175d8d7b51dda9cc57971feR326-R349)
* Minor logic tweaks, such as removal of unnecessary filter assignments. (`conf/modules.config` [conf/modules.configL253](diffhunk://#diff-bf809c928cb10b54d251dec8a140a2d5505150f97175d8d7b51dda9cc57971feL253))

These changes collectively improve the clarity, usability, and maintainability of the pipeline's configuration and outputs.

### CODE REVIEW

**Comment by FerriolCalvet** [comment]
update filter to flag

File: assets/useful_scripts/signatures_sigprofilerextractor.py [hunk 1]
```diff
@@ -35,4 +35,6 @@ def main(input_type, output_dir, input, ref_genome, max_sig, nmf_replicates, cpu
 if __name__ == "__main__":
     main()
 
-### EXAMPLE OF COMMAND: python3 sigprofiler_extractor2.py matrix ./output_dir count_matrix_wgs_20240821.txt --ref_genome GRCh38 --max_sig 10 --nmf_replicates 100 --cpu 10
\ No newline at end of file
+### EXAMPLE OF COMMAND:
+# python3 sigprofiler_extractor2.py matrix ./output_dir count_matrix_wgs_20240821.txt --ref_genome GRCh38 --max_sig 10 --nmf_replicates 100 --cpu 10
+# This can be run using the deepCSA output available at: processing_files/matrixconcatwgs/samples_matrix.*sp.round.tsv
```
Comment (m-huertasp) [inline assets/useful_scripts/signatures_sigprofilerextractor.py]: Maybe we can move this to the Docstring above?


```suggestion
"""
Run SigProfilerExtractor with the specified parameters.
This script can be run using the deepCSA output available at: 
processing_files/matrixconcatwgs/samples_matrix.*sp.round.tsv

Usage:
----------
python3 sigprofiler_extractor2.py
    matrix \
    ./output_dir \
    count_matrix_wgs_20240821.txt \
    --ref_genome GRCh38 \
    --max_sig 10 \
    --nmf_replicates 100 \
    --cpu 10
"""
```

File: bin/check_samplesheet.py [hunk 2]
```diff
@@ -38,36 +38,30 @@ def __init__(
         sample_col="sample",
         vcf_col="vcf",
         bam_col="bam",
-        pileupbam_col="pileup_bam",
-        pileupind_col="pileup_ind",
         **kwargs,
     ):
         """
         # TODO: update this docstring
```
Comment (m-huertasp) [inline bin/check_samplesheet.py]: Could we remove the TODO then?

File: bin/compare_trinucleotide_proportions.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+import click
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+import numpy as np
+
+
+def plot_trinucleotide_proportions(wgs_counts_file):
+
```
Comment (m-huertasp) [inline bin/compare_trinucleotide_proportions.py:15]:
```suggestion
"""
This function generates two scatter plots showing how trinucleotide contexts are distributed across
WGS versus four consensus panels: all, non_protein_affecting, introns_intergenic, and exons_splice_sites.

Args:
    wgs_counts_file (str): Path to tab-separated file containing WGS trinucleotide context counts.
    
"""
```

If you think you don't need it, it is not super important in this file but could be useful.

File: bin/merge_annotation_depths.py [hunk 4]
```diff
@@ -25,10 +25,16 @@ def annotate_depths(annotation_file, depths_file, json_f, input_file):
...
+    if 'bam' in input_csv.columns:
+        bam2sample_dict = dict(zip(input_csv["bam"].astype(str).apply(lambda x: x.split("/")[-1]),
+                                   input_csv["sample"].astype(str)
+                                   )
+                                )
+    else:
+        bam2sample_dict = dict(zip(input_csv["sample"].astype(str),
+                                   input_csv["sample"].astype(str)
+                                   )
+                                )
```
Comment (m-huertasp) [inline bin/merge_annotation_depths.py]: I've tried this and the result is the same but avoiding the lambda. Use it if you want.
```suggestion
    if 'bam' in input_csv.columns:
        bam_names = input_csv["bam"].astype(str).str.split("/").str[-1]
        bam2sample_dict = dict(zip(bam_basenames, input_csv["sample"].astype(str)))

    else:
        samples = input_csv["sample"].astype(str)
        bam2sample_dict = dict(zip(samples, samples))
    
```

File: bin/mutprof_2compute_trinucleotide.py [hunk 5]
```diff
@@ -41,26 +44,71 @@ def compute_trinucleotides(sample_name, depths_file, pseudocount = 0):
                                                                 header = True,
                                                                 index = False)
 
+    return trinucleotides_per_sample[["CONTEXT", sample_name]]
+
+
+def plot_trinucleotide_proportions(wgs_counts_file, sample_trinucleotides, sample_name):
```
Comment (m-huertasp) [inline bin/mutprof_2compute_trinucleotide.py:50]: I think that if you refactor a bit the function in `compare_trinucleotide_proportions.py` and separate it in parts, you could reuse it here but maybe it is too much work

File: bin/write_mafs.py [hunk 6]
```diff
@@ -16,6 +16,15 @@ def main(maf_file, groups_json):
     with open(groups_json, 'r') as file:
         groups_info = json.load(file)
 
+    all_samples = set([item for sublist in groups_info.values() for item in sublist])
+    available_samples = set(maf_df["SAMPLE_ID"].unique())
```
Comment (m-huertasp) [inline bin/write_mafs.py]: I think that when using the `set()` you are already removing duplicates so there's no need to use both `set()`and `unique()` unless you need the items ordered.

File: conf/tools/omega.config [hunk 7]
```diff
@@ -158,7 +158,7 @@ process {
         publishDir = [
             [
                 mode: params.publish_dir_mode,
-                path: { "${params.outdir}/omegagloballoc/plot" },
+                path: { "${params.outdir}/selection/omegagloballoc/plot" },
```
Comment (m-huertasp) [inline conf/tools/omega.config]: This is just a suggestion, now that I see it. There is an output folder named `plots`, why not having all the plots there? I don't know eh maybe you have it clear and this is fine for me as well.

File: conf/tools/oncodrive3d.config [hunk 8]
```diff
@@ -50,7 +50,7 @@ process {
         publishDir = [
             [
                 mode: params.publish_dir_mode,
-                path: { "${params.outdir}/oncodrive3d/plot" },
+                path: { "${params.outdir}/selection/oncodrive3d/plot" },
```
Comment (m-huertasp) [inline conf/tools/oncodrive3d.config]: Same as with the omega plots, but I am okay with whatever you decide

File: conf/modes/clonal_structure.config [hunk 9]
```diff
@@ -17,9 +17,9 @@ params {
     site_comparison_grouping = 'all'
     omega_plot               = true
 
-    omega_withingene         = true
-    omega_autodomains        = true
-    omega_autoexons          = true
+    create_subgenic_regions  = true
+    autodomains             = true
+    autoexons               = true
```
Comment (m-huertasp) [inline conf/modes/clonal_structure.config]: es una tonteria this but you say ja que estic
```suggestion
    autodomains              = true
    autoexons                = true
```

File: conf/modules.config [hunk 10]
```diff
@@ -571,7 +574,7 @@ process {
     withName: MULTIQC {
         ext.args   = params.multiqc_title ? "--title \"${params.multiqc_title}\"" : ''
         publishDir = [
-            path: { "${params.outdir}/multiqc" },
+            path: { "${params.outdir}/processing_files/multiqc" },
```
Comment (m-huertasp) [inline conf/modules.config:577]: This is a gustos but this could also be in the `qc` folder? Or maybe not because in the `qc` you only add like "internal" qcs.

File: conf/modules.config [hunk 11]
```diff
@@ -622,6 +625,10 @@ process {
                     ? 'mm39'
                     : null
     }
+
+    withLabel : deepcsa_core {
+        container = "docker.io/bbglab/deepcsa-core:0.0.2-alpha"
```
Comment (m-huertasp) [inline conf/modules.config:630]: nice!

File: modules/local/compute_mutability/main.nf [hunk 12]
```diff
@@ -33,7 +28,7 @@ process COMPUTE_RELATIVE_MUTABILITY {
                     --depths ${depths} \\
                     --profile ${mut_profile} \\
                     --bedfile ${bedfile} \\
-                    --out_mutability ${prefix}.relative_mutability_per_site.tsv \\
+                    --out_mutability ${prefix} \\
```
Comment (m-huertasp) [inline modules/local/compute_mutability/main.nf:31]: I like that the name is now in the python script instead of here!

File: subworkflows/local/omega/main.nf [hunk 13]
```diff
@@ -161,21 +161,21 @@ workflow OMEGA_ANALYSIS{
...
+    site_comparison_results.map { it -> 
         def meta = it[0]
         def all_files = it[1..-1].flatten()
         [meta, all_files]
     }.set{ site_comparison_results_flattened }
 
 
     ESTIMATOR.out.results.map{ it -> it[1]}.flatten().set{ all_indv_results }
-    all_indv_results.collectFile(name: "all_omegas${suffix}.tsv", storeDir:"${params.outdir}/omega", skip: 1, keepHeader: true).set{ all_results }
+    all_indv_results.collectFile(name: "all_omegas${suffix}.tsv", storeDir:"${params.outdir}/selection/omega", skip: 1, keepHeader: true).set{ all_results }
```
Comment (m-huertasp) [inline subworkflows/local/omega/main.nf:178]: By doing this, would it work with the `-resume`? Also, I think the `set{}` might not be needed, althouth I am not 100% sure.

File: tower.yml [hunk 14]
```diff
@@ -0,0 +1,3 @@
+reports:
+  "plots/mutations_summary*":
+    display: "Mutations summary"
```
Comment (m-huertasp) [inline tower.yml:3]: Is this finally working?

File: subworkflows/local/enrichpanels/main.nf [hunk 15]
```diff
@@ -32,40 +32,44 @@ workflow ENRICHPANELS {
...
         prot_expanded_panel         = prot_consensus_panel
         synonymous_expanded_panel   = synonymous_consensus_panel
         exons_expanded_panel        = exons_consensus_panel
+
+        // FIXME
+        // these are dummy channels that should not be used when
+        // create_subgenic_regions is false check if they can be empty
         all_json_subgenic           = all_bedfile
         nonprot_json_subgenic       = nonprot_bedfile
         prot_json_subgenic          = prot_bedfile
```
Comment (m-huertasp) [inline subworkflows/local/enrichpanels/main.nf]: I've tested and if you change them by `channel.empty()` it works either when `create_subgenic_regions is false or true.
```suggestion
        all_json_subgenic           = channel.empty()
        nonprot_json_subgenic       = channel.empty()
        prot_json_subgenic          = channel.empty()
```

File: workflows/deepcsa.nf [hunk 16]
```diff
@@ -267,22 +258,22 @@ workflow DEEPCSA{
...
-        MUTDENSITYPROT(somatic_mutations, DEPTHSPROTCONS.out.subset, CREATEPANELS.out.prot_consensus_bed, ENRICHPANELS.out.prot_consensus_expanded_panel)
-        MUTDENSITYNONPROT(somatic_mutations, DEPTHSNONPROTCONS.out.subset, CREATEPANELS.out.nonprot_consensus_bed, ENRICHPANELS.out.nonprot_consensus_expanded_panel)
-        MUTDENSITYSYNONYMOUS(somatic_mutations, DEPTHSSYNONYMOUSCONS.out.subset, CREATEPANELS.out.synonymous_consensus_bed, ENRICHPANELS.out.synonymous_consensus_expanded_panel)
+        MUTDENSITYALL(somatic_mutations, DEPTHSALLCONS.out.subset, CREATEPANELS.out.all_consensus_bed, ENRICHPANELS.out.all_consensus_expanded_panel.first())
+        MUTDENSITYPROT(somatic_mutations, DEPTHSPROTCONS.out.subset, CREATEPANELS.out.prot_consensus_bed, ENRICHPANELS.out.prot_consensus_expanded_panel.first())
+        MUTDENSITYNONPROT(somatic_mutations, DEPTHSNONPROTCONS.out.subset, CREATEPANELS.out.nonprot_consensus_bed, ENRICHPANELS.out.nonprot_consensus_expanded_panel.first())
+        MUTDENSITYSYNONYMOUS(somatic_mutations, DEPTHSSYNONYMOUSCONS.out.subset, CREATEPANELS.out.synonymous_consensus_bed, ENRICHPANELS.out.synonymous_consensus_expanded_panel.first())
```
Comment (m-huertasp) [inline workflows/deepcsa.nf]: In the `ENRICHPANELS` subworkflow you already add the `.first()` in the emit block so there's no need to add it again, right?

```suggestion
        MUTDENSITYALL(somatic_mutations, DEPTHSALLCONS.out.subset, CREATEPANELS.out.all_consensus_bed, ENRICHPANELS.out.all_consensus_expanded_panel)
        MUTDENSITYPROT(somatic_mutations, DEPTHSPROTCONS.out.subset, CREATEPANELS.out.prot_consensus_bed, ENRICHPANELS.out.prot_consensus_expanded_panel)
        MUTDENSITYNONPROT(somatic_mutations, DEPTHSNONPROTCONS.out.subset, CREATEPANELS.out.nonprot_consensus_bed, ENRICHPANELS.out.nonprot_consensus_expanded_panel)
        MUTDENSITYSYNONYMOUS(somatic_mutations, DEPTHSSYNONYMOUSCONS.out.subset, CREATEPANELS.out.synonymous_consensus_bed, ENRICHPANELS.out.synonymous_consensus_expanded_panel)
```

**Review by m-huertasp — COMMENTED** [review]
I really like the changes you've added (which are quite a lot). I've been adding some minor coments and I've tested some things, everything looks quite good!

There are more warnings about the `.first()` thing than before, I've been checking and I still don't understand why or how to solve them all without crashing something.

If you want we can discuss about an coment!

File: bin/check_samplesheet.py (hunk 2 above)
Comment (FerriolCalvet) [reply bin/check_samplesheet.py]: yes

File: bin/merge_annotation_depths.py (hunk 4 above)
Comment (FerriolCalvet) [reply bin/merge_annotation_depths.py]: I replaced the bam_names by bam_basenames which I guess was a typo

File: bin/mutprof_2compute_trinucleotide.py (hunk 5 above)
Comment (FerriolCalvet) [reply bin/mutprof_2compute_trinucleotide.py:50]: let's open an issue to refactor the scripts and see what could be shared

File: conf/modules.config (hunk 10 above)
Comment (FerriolCalvet) [reply conf/modules.config:577]: this multiqc is a bit useless so I prefer to keep it in processing_files

File: conf/tools/omega.config (hunk 7 above)
Comment (FerriolCalvet) [reply conf/tools/omega.config]: probably we should move it but this would mean moving the oncodrive3d as well, leaving it for later and also pending to have any other feedback from users

File: subworkflows/local/enrichpanels/main.nf (hunk 15 above)
Comment (FerriolCalvet) [reply subworkflows/local/enrichpanels/main.nf]: thanks for testing it!!

Comment (FerriolCalvet) [reply subworkflows/local/enrichpanels/main.nf]: I have also added the channel.empty for the other two, I guess it is also fine right?

File: subworkflows/local/omega/main.nf (hunk 13 above)
Comment (FerriolCalvet) [reply subworkflows/local/omega/main.nf:178]: I added the set to be able to emit it outside of the subworkflow

File: tower.yml (hunk 14 above)
Comment (FerriolCalvet) [reply tower.yml:3]: not really tested, but I started to add it so that we can test it hahaha

Comment (FerriolCalvet) [reply tower.yml:3]: small fix and now maybe it works

File: workflows/deepcsa.nf (hunk 16 above)
Comment (FerriolCalvet) [reply workflows/deepcsa.nf]: it should not be required, but let's see if it works...

for now I am removing it but we can add a small fix later if needed

**Review by FerriolCalvet — COMMENTED** [review]
thanks for the comments and testing things Marta!!

I am merging it and let's add you changes and we will see if there is any corner case that we missed to test
